syntax = "proto3";

package grpc.service;

import "google/protobuf/timestamp.proto";

enum Status {
    NOT_STARTED = 0;
    IN_PROGRESS = 1;
    COMPLETED = 2;
}

message WorkItem {
    string id = 1;
    string username = 2; 
    string payload = 3;
    Status status = 4;
    bool priority = 5;
    google.protobuf.Timestamp timestamp = 6;
}